### PR TITLE
feat: operators registry module

### PR DESCRIPTION
**Pull request by vgorkavenko**
Now we use lido nestjs module to get lido validators

### CODE REVIEW

**Review by dgusakov — APPROVED** [review]

File: src/inspector/inspector.service.ts [hunk 1]
```diff
@@ -42,14 +42,15 @@ export class InspectorService implements OnModuleInit {
...
             const headEpoch = await this.calculateHeadEpoch();
-            const possibleHighRewardValidators = await this.dataProcessor.getPossibleHighRewardValidatorIndexes(res.lidoIDs, headEpoch);
-            await this.statsProcessor.calculateLidoStats(slotToWrite, possibleHighRewardValidators);
-            await this.statsProcessor.calculateOtherStats(res.otherCounts);
-            await this.statsProcessor.finalizeAppIterate(slotToWrite);
-            await this.criticalAlertService.sendCriticalAlerts(slotToWrite);
+            possibleHighRewardValidators = await this.dataProcessor.getPossibleHighRewardValidatorIndexes(res.lidoIDs, headEpoch);
           }
+          await this.statsProcessor.calculateLidoStats(slotToWrite, possibleHighRewardValidators);
+          await this.statsProcessor.calculateOtherStats(res.otherCounts);
```
Comment (madlabman) [inline src/inspector/inspector.service.ts:51]: Is here a possibility to add these 2 calls to `Promise.all`?

Comment (vgorkavenko) [reply src/inspector/inspector.service.ts:51]: Not sure if this makes sense, but I'll think about it in another task (some optimization, for example)

File: src/inspector/processing/data-processing.service.ts [hunk 2]
```diff
@@ -62,20 +62,20 @@ export class DataProcessingService implements OnModuleInit {
...
           const slotRes = await fetcherWriter.fetchSlotData();
           otherCounts = await fetcherWriter.writeSlotData(slotRes);
           lidoIDs = await this.storage.getLidoValidatorIDs(slotToWrite);
           const epochRes = await fetcherWriter.fetchEpochData(lidoIDs);
           await fetcherWriter.writeEpochData(lidoIDs, epochRes);
         } else {
-          lidoIDs = await this.storage.getLidoValidatorIDs(this.latestSlotInDb);
           const [slotRes, epochRes] = await Promise.all([fetcherWriter.fetchSlotData(), fetcherWriter.fetchEpochData(lidoIDs)]);
           [otherCounts] = await Promise.all([fetcherWriter.writeSlotData(slotRes), fetcherWriter.writeEpochData(lidoIDs, epochRes)]);
         }
```
Comment (madlabman) [inline src/inspector/processing/data-processing.service.ts:81]: I think we can leave `lidoIDs = await this.storage.getLidoValidatorIDs(slotToWrite);`
in the `if` statement body and to place the rest right after:

```js
const [slotRes, epochRes] = await Promise.all([fetcherWriter.fetchSlotData(), fetcherWriter.fetchEpochData(lidoIDs)]);
[otherCounts] = await Promise.all([fetcherWriter.writeSlotData(slotRes), fetcherWriter.writeEpochData(lidoIDs, epochRes)]);
```

File: src/inspector/processing/stats-processing.service.ts [hunk 3]
```diff
@@ -29,19 +30,19 @@ export class StatsProcessingService implements OnModuleInit {
    * Calc stats by storage (validators info, att and prop duties) and push them to Prometheus
    */
   async calculateLidoStats(slot: bigint, possibleHighRewardValidators: string[]): Promise<void> {
-    const operators = await this.nodeOperatorsContract.getOperators();
+    const operators = await this.registryService.getOperators();
     this.prometheus.contractKeysTotal.set(
       { type: 'total' },
-      operators.reduce((sum, o) => sum + o.totalSigningKeys.toNumber(), 0),
+      operators.reduce((sum, o) => sum + o.totalSigningKeys, 0),
```
Comment (madlabman) [inline src/inspector/processing/stats-processing.service.ts:36]: `o.totalSigningKeys` is `number` now or just rely on type casting?

File: src/validators/registry/registry.service.ts [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+   * Collects updates from the registry contract and saves the changes to the database
+   */
+  public async updateValidators(): Promise<void> {
+    await this.prometheusService.trackTask('update-validators', async () => {
+      try {
+        await this.validatorService.update('latest');
+      } catch (error) {
+        // Here we can get a timeout error or something else
+        this.logger.warn('Failed to update validators');
+        const curr = await this.keyStorageService.findUsed();
```
Comment (madlabman) [inline src/validators/registry/registry.service.ts:43]: What if this call fails as well?

File: src/inspector/processing/stats-processing.service.ts (hunk 3 above)
Comment (vgorkavenko) [reply src/inspector/processing/stats-processing.service.ts:36]: It's `number` according to entity description

File: src/inspector/processing/data-processing.service.ts (hunk 2 above)
Comment (vgorkavenko) [reply src/inspector/processing/data-processing.service.ts]: done

File: src/validators/registry/registry.service.ts (hunk 4 above)
Comment (vgorkavenko) [reply src/validators/registry/registry.service.ts:43]: Added handling where it makes sense

File: src/inspector/processing/data-processing.service.ts (hunk 2 above)
Comment (vgorkavenko) [reply src/inspector/processing/data-processing.service.ts:82]: reverted

**Review by madlabman — APPROVED** [review]

**Review by madlabman — APPROVED** [review]

**Pull request merged**